presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTemporalFunction.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.Type;
+
+import java.time.Duration;
+
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static java.util.Objects.requireNonNull;
+
+public class RaptorTemporalFunction
+{
+    private final Type temporalColumnType;
+
+    public RaptorTemporalFunction(Type temporalColumnType)
```

Comment:
Rather than check every time, use a factory method that returns the correct function for the type